src/PriceCalculatorKata/Product.cs

```diff
@@ -40,11 +42,30 @@ public void CheckPercentageValidation(double value)
         }
         public void ReportProductPrice()
         {
-            double DiscountAmount = Math.Round(Price * DiscountPercentage, 2);
-            double PriceWithTaxAndDiscount = Math.Round(Price * (1 + TaxPercentage - DiscountPercentage), 2);
-            Console.WriteLine($"${Price} before tax and discount and ${PriceWithTaxAndDiscount} after {TaxPercentage * 100}% tax and {DiscountPercentage * 100}% discount");
+            double DiscountAmount = GetTotalDiscountAmount();
+            double TaxAmount = Price * TaxPercentage;
+            double PriceWithTaxAndDiscount = Math.Round(Price + TaxAmount - DiscountAmount, 2);
+            Console.WriteLine($"${Price} before tax and discount and ${PriceWithTaxAndDiscount} after {TaxPercentage * 100}% tax and {UniversalDiscountPercentage * 100}% discount");
             if (DiscountAmount > 0)
-                Console.WriteLine($"Discount amount = {DiscountAmount}");
+                Console.WriteLine($"Total Discount amount = {DiscountAmount}");
+        }
+        private double GetTotalDiscountAmount()
```

Comment:
product class getting bigger. better move out these methods to separate classes and extensions.